File: app/src/main/java/com/example/huzheyuan/scout/QRCode/DensityUtils.java
package com.example.huzheyuan.scout.qRCode;

/**
 * Created by leo428 on 1/16/17.
 */
import android.content.Context;

public class DensityUtils {


    /**
     * px与dp互相转换
     * Created by yangle on 2016/4/12.
     */


    public static int dp2px(Context context, float dp) {
        //获取设备密度
        float density = context.getResources().getDisplayMetrics().density;
        //4.3, 4.9, 加0.5是为了四舍五入
        int pX = (int) (dp * density + 0.5f);
        return pX;
    }

    public static float px2dp(Context context, int px) {
        //获取设备密度
        float density = context.getResources().getDisplayMetrics().density;
        float dP = px / density;
        return dP;
    }

}

File: app/src/main/java/com/example/huzheyuan/scout/utilities/DensityUtils.kt
package com.example.huzheyuan.scout.utilities

/**
 * Created by leo428 on 1/16/17.
 * Credit for Eddie for doing nothing
 */
import android.content.Context

class DensityUtils(context: Context) {
    /**
     * px与dp互相转换
     */
    var density = 0f
    init {
        density = context.resources.displayMetrics.density //获取设备密度
    }
    fun dp2px(dp: Float): Int {
        //加0.5是为了四舍五入
        val pX = (dp * density + 0.5f).toInt()
        return pX
    }

    fun px2dp(px: Int): Float {
        val dP = px / density
        return dP
    }
}
